cli/tauri-cli/src/bundle/wix.rs

```diff
@@ -448,3 +508,74 @@ fn generate_external_binary_data(settings: &Settings) -> crate::Result<Vec<Exter
 
   Ok(external_binaries)
 }
+
+type ResourceMap = BTreeMap<String, ResourceDirectory>;
+// generates the data required for the resource bundling on wix
+fn generate_resource_data(settings: &Settings) -> crate::Result<ResourceMap> {
+  let mut resources = ResourceMap::new();
+  let regex = Regex::new(r"[^\w\d\.]")?;
+  let cwd = std::env::current_dir()?;
+  for src in settings.resource_files() {
+    let src = src?;
+
+    let filename = src
+      .file_name()
+      .expect("failed to extract resource filename")
+      .to_os_string()
+      .into_string()
+      .expect("failed to convert resource filename to string");
+
+    let resource_path = cwd
+      .join(src.clone())
+      .into_os_string()
+      .into_string()
+      .expect("failed to read resource path");
+
+    let resource_entry = ResourceFile {
+      guid: generate_guid(filename.as_bytes()).to_string(),
+      path: resource_path,
+      id: regex.replace_all(&filename, "").to_string(),
+    };
+
+    // split the resource path directories
+    let mut directories = src.components().filter(|component| {
+      let comp = component.as_os_str();
+      comp != "." && comp != ".."
+    }).collect::<Vec<_>>();
+    directories.truncate(directories.len() - 1);
+    
+    // transform the directory structure to a chained vec structure
+    for directory in directories {
+      let directory_name = directory.as_os_str().to_os_string().into_string().expect("failed to read resource folder name");
+
+      // if the directory is already on the map
+      if resources.contains_key(&directory_name) {
+        let directory_entry = &mut resources.get_mut(&directory_name).unwrap();
```

Comment:
get rid of this `unwrap()`.  Should either be a `?` if is a result or an `expect()` if is a Option.

Reply:
so I should always use expect, even when I know the result is Ok()? In this case I've already checked if the key exists.

Reply:
Even if you know it exists, its a bit more expressive that way.  And in the case where you are using a result, you could use a `?` instead of an expect to bubble the error upwards.